File: 2807 Insert Greatest Common Divisors in Linked List/2807.py
class Solution:
    def insertGreatestCommonDivisors(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cur = head

        while cur and cur.next:
            p = ListNode(gcd(cur.val, cur.next.val))
            p.next = cur.next
            cur.next = p
            cur = p.next

        return head

File: 2807 Insert Greatest Common Divisors in Linked List/2807.cpp
class Solution {
public:
    ListNode* insertGreatestCommonDivisors(ListNode* head) {
        ListNode* cur = head;

        while (cur && cur->next){
            ListNode* p = new ListNode(gcd(cur->val, cur->next->val));
            p->next = cur->next;
            cur->next = p;
            cur = p->next;
        }

        return head;
    }
};
